### PR TITLE
Application general delete option bug v4.3

### PR DESCRIPTION
**Pull request by jansobczynski**
## Issue

https://gravitee.atlassian.net/browse/APIM-4183

## Description

Add option to delete application

## Additional context

Screenshot:
Screenshot 2024-03-19 at 1 30 11 PM

🏗️ Your changes can be tested here and will be available soon:
      Console: [https://pr.team-apim.gravitee.dev/6963/console](https://pr.team-apim.gravitee.dev/6963/console)
      Portal: [https://pr.team-apim.gravitee.dev/6963/portal](https://pr.team-apim.gravitee.dev/6963/portal)
      Management-api: [https://pr.team-apim.gravitee.dev/6963/api/management](https://pr.team-apim.gravitee.dev/6963/api/management)
      Gateway v4: [https://pr.team-apim.gravitee.dev/6963](https://pr.team-apim.gravitee.dev/6963)
      Gateway v3: [https://pr.gateway-v3.team-apim.gravitee.dev/6963](https://pr.gateway-v3.team-apim.gravitee.dev/6963)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-ydylhsvgky.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/application/details/general/application-general.component.html [hunk 1]
```diff
@@ -153,5 +153,19 @@ <h3>OpenID Connect Integration</h3>
     </mat-card>
   </ng-container>
 
+  <mat-card class="danger-card">
```
Comment (jourdiw) [inline gravitee-apim-console-webui/src/management/application/details/general/application-general.component.html]: If the user cannot perform any actions in the danger zone, the card should not be shown.

File: gravitee-apim-console-webui/src/management/application/details/general/application-general.component.ts [hunk 2]
```diff
@@ -159,4 +165,35 @@ export class ApplicationGeneralComponent implements OnInit {
       )
       .subscribe(() => this.ngOnInit());
   }
+
+  deleteApplication() {
+    this.matDialog
+      .open<GioConfirmAndValidateDialogComponent, GioConfirmAndValidateDialogData>(GioConfirmAndValidateDialogComponent, {
+        width: '500px',
```
Comment (jourdiw) [inline gravitee-apim-console-webui/src/management/application/details/general/application-general.component.ts]: You can use the particles value: `GIO_DIALOG_WIDTH.MEDIUM`

File: gravitee-apim-console-webui/src/management/application/details/general/application-general.component.ts [hunk 3]
```diff
@@ -159,4 +165,35 @@ export class ApplicationGeneralComponent implements OnInit {
...
+        data: {
+          title: `Delete Application`,
+          content: `Are you sure you want to delete the Application?`,
+          confirmButton: `Yes, delete it`,
+          validationMessage: `Please, type in the name of the api <code>${this.initialApplication.name}</code> to confirm.`,
```
Comment (jourdiw) [inline gravitee-apim-console-webui/src/management/application/details/general/application-general.component.ts]:
```suggestion
          validationMessage: `Please, type in the name of the application <code>${this.initialApplication.name}</code> to confirm.`,
```

Comment (jansobczynski) [reply gravitee-apim-console-webui/src/management/application/details/general/application-general.component.ts]: Done

File: gravitee-apim-console-webui/src/management/application/details/general/application-general.component.ts (hunk 2 above)
Comment (jansobczynski) [reply gravitee-apim-console-webui/src/management/application/details/general/application-general.component.ts]: Done

File: gravitee-apim-console-webui/src/management/application/details/general/application-general.component.html (hunk 1 above)
Comment (jansobczynski) [reply gravitee-apim-console-webui/src/management/application/details/general/application-general.component.html]: Done

**Review by jourdiw — APPROVED** [review]

**Pull request merged**